api/config/v1alpha1/envoygateway_types.go

```diff
@@ -135,6 +156,56 @@ type RateLimitRedisSettings struct {
 	URL string `json:"url"`
 }
 
+// Extension defines the configuration for registering an extension to
+// the Envoy Gateway control plane.
+type Extension struct {
+	// Resources defines the set of K8s resources the extension will handle.
+	Resources []*GroupVersionKind `json:"resources,omitempty"`
+
+	// Service defines the configuration of the extension service that the Envoy
+	// Gateway Control Plane will call through extension hooks.
+	Service *ExtensionService `json:"service"`
+}
+
+// ExtensionService defines the configuration for connecting to a registered extension service.
+type ExtensionService struct {
+	// Host define the extension service hostname.
+	Host string `json:"host"`
+
+	// Port defines the port the extension service is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=80
+	Port int32 `json:"port,omitempty"`
+
+	// TLS defines TLS configuration for communication between Envoy Gateway and
+	// the extension service.
+	//
+	// +optional
+	TLS *ExtensionTLS `json:"tls,omitempty"`
+}
+
+// ExtensionTLS defines the TLS configuration when connecting to an extension service.
+type ExtensionTLS struct {
+	// Type is the method for how the TLS certificate is loaded. Supported types are:
+	//
+	//   * Secret: Load the TLS certificate from a K8s secret.
+	//
+	// +unionDiscriminator
+	Type TLSType `json:"type"`
+
+	// Secret defines which K8s secret to load the TLS certificate from.
+	//
+	// +optional
+	Secret *TLSSecret `json:"secret,omitempty"`
+
+	// File defines the configuration for loading the TLS certificate from the filesystem.
```

Comment:
is this needed as we start off ?

Reply:
for the initial iteration I would say no. I was planning to update the `config.Validate` method to flag if this is set and say it's not supported yet similar to what we do now for `FileProvider`